lib/spack/docs/build_systems/rpackage.rst

```diff
@@ -124,6 +126,67 @@ If you only specify the URL for the latest release, your package will
 no longer be able to fetch that version as soon as a new release comes
 out. To get around this, add the archive directory as a ``list_url``.
 
+^^^^^^^^^^^^^^^^^^^^^
+Bioconductor packages
+^^^^^^^^^^^^^^^^^^^^^
+
+Bioconductor packages are setup in a similar way to CRAN packages, but there
+are some very important distinctions. Bioconductor packages can be found at:
+https://bioconductor.org/. Bioconductor packages are R packages and so follow
+the same packaging scheme as CRAN packages. What is different is that
+Bioconductor itself is versioned and released. This scheme, using the
+Bioconductor package installer, allows further specification of the minimum
+version of R as well as further restrictions on the dependencies between
+packages than what is possible with the native R packaging system. Spack can
+not replicate these extra features and thus Bioconductor packages in Spack need
+to be managed as a group during updates in order to maintain package
+consistency with Bioconductor itself.
+
+Another key difference is that, while previous versions of packages are
+available, they are not available from a site that can be programmatically set,
+thus a ``list_url`` attribute can not be used. However, each package is also
+available in a git repository, with branches corresponding to each Bioconductor
+release. Thus, it is always possible to retrieve the version of any package
+corresponding to a Bioconductor release simply by fetching the branch that
+corresponds to the Bioconductor release of the package repository. For this
+reason, spack Bioconductor R packages use the git repository, with the commit
+of the respective branch used in the ``version()`` attribute of the package.
+
+^^^^^^^^^^^^^^^^^^^^^^^^
+cran and bioc attributes
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Much like the ``pypi`` attribute for python packages, due to the fact that R
+packages are obtained from specific repositories, it is possible to setup shortcut
+attributes that can be used to set ``homepage``, ``url``, ``list_url``, and
+``git``. For example, the following ``cran`` attribute:
+
+.. code-block:: python
+
+   cran = 'caret/caret_6.0-86.tar.gz'
```

Comment:
The reason that the `pypi` attribute has this structure is because:

1. The package name may not always match the tarball name (e.g. `Foo_Bar/foo-bar-1.2.3.tar.gz`)
2. The package may not always use tar (e.g. `foo-bar/foo-bar-1.2.3.zip`)

Is this also the case for CRAN? If not, perhaps we could make this simpler and users would only need to set `cran = 'caret'`.

Reply:
I could be missing something but I think the form with the tarfile is needed for setting the `url` attribute. The `bioc` attribute only needs to set the `git` attribute so it uses the simpler form.

Reply:
Yes, but if a tarball is always available, you could simply set `cran = 'caret'` and set the url to a fake version like `{cran}_1.2.3.tar.gz`.

Reply:
There are some differences in the way r- packages specify versions though. Some variants:
```
{cran}_1.2.3
{cran}_1.2-3
{cran}_1-2
{cran}_1.2
{cran}_1.2.3-1
{cran}_1.02
{cran}_1.2.3.4
{cran}_1.2-3.4
{cran}_1.2.3.4.5
```
There are also some that use date format versions. That does not seem like it could be faked.

Reply:
But that part doesn't matter, as long as Spack detects the version in the URL and replaces it.

Reply:
Okay; I underestimated Spack's power in version processing then. It looks like all of the CRAN packages are `tar.gz` files so I will push out a change later today.

Thanks for pointing that out.

Reply:
Just need to update this line of the docs now.

Reply:
Sorry, I missed that. Should the url example be left as-is or changed to reflect the stand-in version?
```
.. code-block:: python

   cran = 'caret'

is equivalent to:

.. code-block:: python

   homepage = 'https://cloud.r-project.org/package=caret'
   url      = 'https://cloud.r-project.org/src/contrib/caret_6.0-86.tar.gz'
   list_url = 'https://cloud.r-project.org/src/contrib/Archive/caret'
```

Reply:
I think it makes sense to leave it as-is. That makes it consistent with the format that is presented with `spack info`.